src/thd_engine_adaptive.cpp

```diff
@@ -846,7 +846,7 @@ int cthd_engine_adaptive::verify_condition(struct condition condition) {
 		return 0;
 
 	cond_name = condition_names[MIN(MAX(0, condition.condition), G_N_ELEMENTS(condition_names) - 1)];
-	thd_log_error("Unsupported condition %d (%s)\n", condition.condition, cond_name);
+	thd_log_error("Unsupported condition %lu (%s)\n", condition.condition, cond_name);
```

Comment:
Better use `%" PRIu64 "` instead of `%lu` to avoid issues/warnings as the definition of a `long int` depends on the platform.